lib/onebox/engine/wikimedia_onebox.rb

```diff
@@ -0,0 +1,38 @@
+module Onebox
+  module Engine
+    class WikimediaOnebox
+      include Engine
+
+      matches_regexp /^https?:\/\/commons\.wikimedia\.org\/wiki\/(File:.+)/
+      always_https
+
+      def self.priority
+        1
+      end
+
+      def to_html
+        begin
+          match ||= @url.match(/^https?:\/\/commons\.wikimedia\.org\/wiki\/(?<name>File:.+)/)
+          name = match[:name]
+          api_url = "https://en.wikipedia.org/w/api.php?action=query&titles=#{name}&prop=imageinfo&iilimit=50&iiprop=timestamp|user|url&iiurlwidth=500&format=json"
+          result ||= ::JSON.load(open(api_url))
```

Comment:
Engines already handle this, all you need to do is define a `url` method, see e.g.:

https://github.com/discourse/onebox/blob/master/lib/onebox/engine/xkcd_onebox.rb